src/models/Factor.js

```diff
@@ -204,28 +205,30 @@ function (Okta, CookieUtil, factorUtil, BaseLoginModel) {
 
         var promise;
         // MFA_REQUIRED
-        if (transaction.getFactorById) {
-          promise = transaction
-          .getFactorById(this.get('id'))
-          .verifyFactor(data);
+        if (transaction.status === 'MFA_REQUIRED') {
+          var factor = _.findWhere(transaction.factors, {
+            id: this.get('id')
+          });
+          promise = factor.verify(data);
         }
 
         // MFA_CHALLENGE
-        else if (this.get('canUseResend') && transaction.resendByName) {
-          var firstLink = transaction.response._links.resend[0];
-          promise = transaction.resendByName(firstLink.name);
+        else if (this.get('canUseResend') && transaction.resend) {
+          var firstLink = transaction.data._links.resend[0];
+          promise = transaction.resend(firstLink.name);
```

Comment:
Unfortunately not. The `RECOVERY_CHALLENGE` state will only ever have one option to resend, so the argument to resend in the RecoveryChallengeController was ignored. We used to have some logic to handle the discrepancy in the SDK, but it didn't make sense to add it back with the autogen'd methods.

In the `MFA_ENROLL_ACTIVATE` state, we have an array of options to resend to, so a name is required.